src/Microsoft.AspNet.Mvc.Core/Controller.cs

```diff
@@ -376,6 +376,11 @@ public virtual ContentResult Content(string content, string contentType, Encodin
         [NonAction]
         public virtual JsonResult Json(object data)
         {
+            var disposableValue = data as IDisposable;
+            if (disposableValue != null)
+            {
+                Response.OnResponseCompleted(_ => disposableValue.Dispose(), null);
```

Comment:
Out of curiosity, what's the runtime behavior when the Dispose throws? Is that something we should care about?